platform/ui/src/hooks/useMedia.js

```diff
@@ -17,36 +33,37 @@ import { useState, useEffect } from 'react';
  * @returns
  */
 function useMedia(queries, values, defaultValue) {
-  // Array containing a media query list for each query
-  const mediaQueryLists = queries.map(q => window.matchMedia(q));
+  const memoizedQueryMap = useMemo(() => getMediaQueryMap(queries), [queries]);
+  const memoizedValue = useMemo(
+    () => getValue(memoizedQueryMap, values, defaultValue),
+    [mediaQueryLists, values]
+  );
+  // State and setter for matched value
+  const [mediaQueryLists, setMediaQueryLists] = useState(memoizedQueryMap);
+  const [value, setValue] = useState(memoizedValue);
+
+  const mediaQueryListener = () => {
+    const nextValue = getValue(mediaQueryLists, values, defaultValue);
+    setValue(nextValue);
+  };
 
-  // Function that gets value based on matching media query
-  const getValue = () => {
-    // Get index of first media query that matches
-    const index = mediaQueryLists.findIndex(mql => mql.matches);
+  const addMediaQueryListeners = mediaQueryLists => {
+    // Set a listener for each media query with above mediaQueryListener as callback.
+    mediaQueryLists.forEach(mql => mql.addListener(mediaQueryListener));
+  };
 
-    // Return related value or defaultValue if none
-    return typeof values[index] !== 'undefined' ? values[index] : defaultValue;
+  const removeMediaQueryListeners = (mediaQueryLists = []) => {
+    mediaQueryLists.forEach(mql => mql.removeListener(mediaQueryListener));
```

Comment:
Ah. Good call on adding/removing. I could see this causing issues if these weren't removed as we navigate routes. What currently triggers clean-up?